stock_release_channel_partner_public_holidays/models/stock_release_channel.py

```diff
@@ -0,0 +1,39 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class StockReleaseChannel(models.Model):
+
+    _inherit = "stock.release.channel"
+
+    exclude_public_holidays = fields.Boolean()
+
+    def filter_release_channel(self, partner):
+        channels = self
+        for channel in self:
+            if not channel.exclude_public_holidays:
+                continue
+            if channel._is_shipment_date_a_public_holiday(partner):
+                channels -= channel
+        return channels
+
+    def _is_shipment_date_a_public_holiday(self, partner):
+        """
+        Returns True if shipment_date is a public holiday
+        :return: bool
+        """
+        self.ensure_one()
```

Comment:
there's no need to call ensure_one for a private method that is used only in this loop